### DISCUSSION TITLE
Is the data always in memory even if we create db with path?

### DISCUSSION THREAD

**Discussion by zhangxinyu1** [duckdb/duckdb#3936] [Q&A]
If so, is the amount of the data in duckdb is limited with the size of memory?
Or, does duckdb clear a part of the data in memory after the data is flushed?

**Comment by Mytherin** [comment]
If DuckDB is opened with a persistent database connection data from the base tables will be loaded and unloaded from disk as required. This behavior is dictated by the `memory_limit`, which can be set as follows: `SET memory_limit='100GB'`. This defaults to 75% of the physically installed RAM on your machine.

Note that there are several limitations with here still, for example indexes (including primary key and unique constraint indexes) do not count towards this memory limit and cannot be offloaded to disk yet. This is an area that is actively being worked on and improved.

**Reply by zhangxinyu1** [reply]
Thanks. If DuckDB is opened with a persistent database and I keep inserting data, will a part of the data be removed from memory when memory usage reaches the memory_limit?
